server/src/main/java/io/druid/segment/indexing/Transform.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.indexing;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name = "expression", value = ExpressionTransform.class)
+})
+public interface Transform
+{
+  String getName();
```

Comment:
do we really need to have this method ?

Reply:
Yes, it's like the name of an aggregator, you need to have it so we know what the field should be called. If you want a field `z` that is `x + y` then it would be specified like,

```
"transformSpec" : {
  "transforms" : [
    { "type" : "expression", "name" : "z", "expression" : "x + y" }
  ]
}
```

Reply:
ah, i see, its name of the column in the row ... would be nice to add little doc on this to be clear for other implementers.

Reply:
I added javadocs to the interface.